src/core/components/ui/WalletBinding/index.vue

```diff
@@ -0,0 +1,191 @@
+      wallet: (state) => state.substrate.wallet,
+      metamaskWalletAddress: (state) => state.metamask.metamaskWalletAddress,
+    }),
+  },
+  watch: {
+    show() {
+      if (this.show) {
+        this.error = "";
+        this.loading = false;
+        this.putAccount = false;
+        this.putWallet = true;
+        this.inputPassword = false;
+        this.isLoading = false;
+      }
+    },
+  },
+  mounted() {},
+  methods: {
+    ...mapMutations({
+      setMetamaskAddress: "metamask/SET_WALLET_ADDRESS",
+    }),
+    async getMunnyFromFaucet(address) {
+      await getEthFromFaucet(address);
+      await getDaicFromFaucet(address);
+    },
+    async setWalllet(walletName) {
+      this.error = "";
+      this.loading = true;
+      switch (walletName) {
+        case "metamask":
+          this.ethAccount = await startApp();
+          if (this.ethAccount.currentAccount == "no_install") {
+            this.error = "Please install MetaMask!";
+          } else {
+            this.accountList = [];
+            for (let i = 0; i < this.ethAccount.accountList.length; i++) {
+              const balance = await getBalanceETH(
+                this.ethAccount.accountList[i]
+              );
+              var lastAddr = this.ethAccount.accountList[i].substr(
+                this.ethAccount.accountList[i].length - 4
+              );
+              let isActive = false;
+              if (
+                this.metamaskWalletAddress == this.ethAccount.accountList[i]
+              ) {
+                isActive = true;
+              }
+              this.accountList.push({
+                address: this.ethAccount.accountList[i],
+                lastAddr: lastAddr,
+                name: "Account " + (i + 1),
+                balance: parseFloat(balance).toFixed(2),
+                active: isActive,
+              });
+
+              try {
+                await setEthAddress(this.api, this.wallet, this.ethAccount.currentAccount);
+                this.setMetamaskAddress(this.ethAccount.currentAccount);
+                this.$emit("status-wallet", {
+                  status: true,
+                  img: "metamask-account-connected.png",
+                });
+                await this.getMunnyFromFaucet(this.ethAccount.currentAccount);
+                this.isLoading = false;
+              } catch (err) {
+                console.log(err.message);
+                this.isLoading = false;
+                this.error = err.message;
+              }
+            }
+            this.putWallet = false;
+            this.putAccount = true;
+          }
+          break;
+      }
+      this.loading = false;
+    },
+    closeDialog() {
+      this._show = false;
+      this.error = "";
+      this.loading = false;
+      this.putWallet = true;
+      this.inputPassword = false;
+    },
+    async copyToClipboard(text) {
+      await navigator.clipboard.writeText(text)
+    }
+  },
+};
+</script>
+
+<style
```

Comment:
Please remove this if you don't need it.

Reply:
removed